### ISSUE TITLE
Elaborate fb2.css

### ISSUE THREAD

**Issue by hius07** [koreader/koreader#6480]
May I propose some enhancement of fb2.css.
https://github.com/koreader/crengine/blob/master/cr3gui/data/fb2.css

1. Remove line 20
`cite p, epigraph p { text-align: left; text-indent: 0px }`
Further settings for both `epigraph` and `cite` are OK and enough (cite should be justified).

2. Remove in line 24 `font-style: italic` for the `stanza` tag.
I would not be happy to read the whole book of poetry in italic. 
And when a poem is cited within the text, it should not be in italic as well (due to the Handbook of typographical rules it might be of a smaller font-size, but it seems not necessary).
![poem-rules-cite](https://user-images.githubusercontent.com/62179190/89622422-b2d42700-d89b-11ea-8cdd-100b515e1914.jpg)

**Comment by poire-z** [comment]
Please ping the russian readers you had some previous FB2 discussions with - because I have no opinion :)
We can add what you decide.

But what I'd like is for you to cleanup/reorganize our fb2.css - and make it as clean as our epub.css:
- remove the elements that are not defined in the FB2 specs - except may be those that you have really found in real FB2 books (if any, dunno if FB2 allows extending its specs).
- group the elements by type: block/inline and similiarities
- add some comments about what they are used for (difference between poem & stanza ? epigraph/cite/style are unknown to me, I know only HTML :)

The errors, missing stuff, uneeded stuff... might then be easier to catch.

**Comment by roshavagarga** [comment]
Additional descriptions, classes and so on can be found in [this github](https://github.com/gribuser/FB3) which covers FictionBook v3.

**Comment by hius07** [comment]
Neither readers nor developers are interested in fb3, sorry.

**Comment by poire-z** [comment]
The bit about developers is a bit untrue :) 
`https://github.com/buggins/coolreader/pull/105`
https://github.com/koreader/crengine/pull/314
https://github.com/koreader/crengine/commit/fc831db602af1a5b4b3c343e6b7c998bdd9042fb#diff-4296cf0f47007e4df3fa206b9eb2dd87
As far as CSS is concerned, I think we just added `<spacing>` and `<underline>` with FB3.

**Comment by hius07** [comment]
Thank you for that, but nobody reads book in fb3.
I've checked the CoolReader and other threads at 4pda.ru, that's the fact.

**Comment by roshavagarga** [comment]
https://www.litres.ru seems to offer fb3 as an alternative to fb2, so there is a chance that somebody does do it, albeit I'll agree it's definitely not as popular.

**Comment by poire-z** [comment]
Btw, there are hacks in the HTML/TXT parser to get better output with books from [Lib.Ru](http://lib.ru/), espacially with these 2 kind of formats I think:
http://lib.ru/INPROZ/ASTURIAS/asturias1_1.txt ("Lib.ru html")
http://lib.ru/INPROZ/ASTURIAS/asturias1_1.txt_with-big-pictures.html ("Fine HTML")

Is it still an active source of russian books ? and I guess I can't just drop that stuff?

**Comment by hius07** [comment]
Yes, lib.ru is good and active source of books.

**Comment by virxkane** [comment]
> Is it still an active source of russian books ? and I guess I can't just drop that stuff?

Yes, it still an active source of russian books. https://en.wikipedia.org/wiki/Lib.ru
But I don't know what percentage of users still use it.

**Comment by poire-z** [comment]
@virxkane : can you post some screenshots on how the first and second pages (with a break between 1st and 2nd real paragraphs) of http://lib.ru/INPROZ/ASTURIAS/asturias1_1.txt look with CR 3.238 ?
I get it in a monospace font because of some PRE left-in (but the paragraphs are disctinct).
If I drop the PRE, I get one big paragraph.
I just want to know how it was "before" I came in :) and how it's expected to look (may be the monospace was already there, and it was considered good enough so I don't have to fix that).
The "Fine HTML" version http://lib.ru/INPROZ/ASTURIAS/asturias1_1.txt_with-big-pictures.html looks better, not in monospace.

**Comment by virxkane** [comment]
> can you post some screenshots on how the first and second pages

Ok, but you can yourself build coolreader, you only need qt5-devel and cmake.
```sh
git clone --depth=1 --branch=cr3.2.38 https://github.com/buggins/coolreader.git coolreader-3.2.38
mkdir coolreader-build
cd coolreader-build
cmake -DCMAKE_BUILD_TYPE=Debug -DCMAKE_INSTALL_PREFIX=/usr/local/cr3.2.38 ../coolreader-3.2.38
make
sudo make install
```
call in via:
```
/usr/local/cr3.2.38/bin/cr3
```

There is some kind of glitch, immediately after loading there is also a monospaced font.
![cr3 2 38-1](https://user-images.githubusercontent.com/36960933/90804308-185bf500-e32b-11ea-8c73-da55e1bb7977.png)

But when I resize the window or open the preferences window and immediately click OK, everything changes:
![cr3 2 38-3](https://user-images.githubusercontent.com/36960933/90805144-4c83e580-e32c-11ea-97e2-c9d8d9f2a78b.png)

**Comment by poire-z** [comment]
OK, thanks. So it was already a mess and same as with current KOReader (and when changing font size or margins, we additionally get a popup telling us styles haves changed and a reload is needed, so I didn't even noticed it was fine below that popup :)
it's quite hacky the way it's done: the text is in a PRE in the document, and it's parsed as PRE - so the pre-text can be splitted into P and converted to nice HTML - and at the end, the PRE gets its node name subtitute with DIV, so that on next renderings it's not considered PRE (because indeed, the pre-text has been replaced by nice HTML)... Still fighting with that :/

> you can yourself build coolreader, you only need qt5-devel and cmake.

I don't have/want anything QT on my little dev linux machine - and I don't want to get more friend with coolreader :) Thanks for the screenshots, they are enough for me to go on.

**Comment by mergen3107** [comment]
>     1. Remove line 20
>        `cite p, epigraph p { text-align: left; text-indent: 0px }`
>        Further settings for both `epigraph` and `cite` are OK and enough (cite should be justified).

I think these should be left as they are, with some modifications. For example, `epigraph` is a block, which we can `text-align: right` (as most epigraphs are in Russian books, AFAIK), while `epigraph p` is the text inside this block. So we can `text-align: left`. See [example](https://nestoriana.wordpress.com/2019/04/30/td_zemelyushka/):

![](https://nestoriana.files.wordpress.com/2019/04/d0a2d094-d0bdd0b0d187d0b0d0bbd0bed09ed09a.jpg)

I am not sure what should we do with `cite`. Should we `text-indent` it by, let's say, more than paragraph's `text-indent`? And maybe justify `cite p` inside it?



>     2\. Remove in line 24 `font-style: italic` for the `stanza` tag.
>         I would not be happy to read the whole book of poetry in italic.

I totally agree with this!
Didn't realize I didn't check poems yesterday when I tweaked my css.

**Comment by hius07** [comment]
I made a set of style tweaks (cite, pre, stanza etc) for myself a year ago and haven't digged into it since then. I'll refresh some ideas.
Meanwhile it's interesting to observe the huge work and discussion on epub.css: https://github.com/koreader/crengine/pull/194
